tests/test_downstream_issue.py

```diff
@@ -1831,6 +1838,173 @@ def test_remove_diacritics(self):
             actual = remove_diacritics(text)
             self.assertEqual(actual, expected)
 
+    def test_get_field_id_by_name(self):
+        """Test _get_field_id_by_name function"""
+        # Clear cache first
+        d.jira_cache.clear()
+
+        mock_client = MagicMock()
+        mock_client.fields.return_value = [
+            {"name": "Story Points", "id": "customfield_12345"},
+            {"name": "Epic Link", "id": "customfield_67890"},
+        ]
+
+        # Test field found - fetches and caches
+        result = d._get_field_id_by_name(mock_client, "Story Points")
+        self.assertEqual(result, "customfield_12345")
+        mock_client.fields.assert_called_once()
+
+        # Test cache - should use cache, not fetch again
+        result = d._get_field_id_by_name(mock_client, "Epic Link")
+        self.assertEqual(result, "customfield_67890")
+        self.assertEqual(mock_client.fields.call_count, 1)
+
+        # Test field not found - will fetch again since not in cache
+        result = d._get_field_id_by_name(mock_client, "Non Existent Field")
+        self.assertIsNone(result)
+        self.assertEqual(mock_client.fields.call_count, 2)
+
+    def test_resolve_field_identifier(self):
+        """Test _resolve_field_identifier function"""
+        # Clear cache first
+        d.jira_cache.clear()
+
+        mock_client = MagicMock()
+        mock_client.fields.return_value = [
+            {"name": "Story Points", "id": "customfield_12345"},
+        ]
```

Comment:
The code in `_resolve_field_identifier()` doesn't reference `d.jira_cache`, and its subroutine, `_get_field_id_by_name()` already has its own unit test, so it doesn't need to be tested here.  That is, instead of maintaining the cache here, we should be mocking `_get_field_id_by_name()` and setting its return value instead of using the client mock.